packages/viewer/src/components/scene-tree/lib/errors.ts

```diff
@@ -29,5 +33,43 @@ function getSceneTreeErrorMessage(code: SceneTreeErrorCode): string {
       return 'Disconnected from server.';
     case SceneTreeErrorCode.SUBSCRIPTION_FAILURE:
       return 'The tree was not able to receive subscription events';
+    case SceneTreeErrorCode.UNAUTHORIZED:
+      return 'The tree was unauthorized to connect to the server. The associated Viewer may not be connected.';
+  }
+}
+
+export class SceneTreeError extends CustomError {
+  public constructor(message: string, e?: Error) {
+    super(message, e);
+
+    // Allows for `instanceof` checks.
+    Object.setPrototypeOf(this, SceneTreeError.prototype);
+  }
+}
+
+export class SceneTreeUnauthorizedError extends SceneTreeError {
+  public constructor(message: string, e?: Error) {
+    super(message, e);
+
+    // Allows for `instanceof` checks.
```

Comment:
that's interesting you have to set this to allow it. I didn't know this.

Reply:
Yeah, definitely an interesting detail! We had to do similar [here](https://github.com/Vertexvis/vertex-web-sdk/blob/master/packages/viewer/src/lib/errors.ts)